ctrlf.el

```diff
@@ -884,18 +895,30 @@ change the search style if already in a search."
     (unless preserve
       (setq ctrlf--style style))
     (if ctrlf--active-p
-        (if (and (not inhibit-history)
-                 (string-empty-p (field-string (point-max))))
-            (previous-history-element 1)
-          (ctrlf-next-match))
+        (cond
+         ;; Continue search with `arg'.
+         ;;
+         ;; TODO: Is there a better way to replace the minibuffer
+         ;; contents? Should we use `(with-current-buffer
+         ;; ctrlf--minibuffer...)' or is this safe?
```

Comment:
I'm very new to manipulating the minibuffer contents like this. This was the best I could find based on research into `previous-history-element`. I've caught myself accidentally inserting over the entire current buffer when my code didn't properly leave the `with-current-buffer` statement before calling `ctrlf-forward`. This would be more of a crutch, and I think correct code should not require it.